CSharpSourceCode/Battle/StatusEffects/StatusEffectMissionLogic.cs

```diff
@@ -28,6 +32,93 @@ public override void OnAgentCreated(Agent agent)
             }
         }
 
+
+        public override void OnAfterMissionCreated()
+        {
+            base.OnAfterMissionCreated();
+
+            //var attacker = Mission.Current.Teams.Attacker;
+            
+            var attacker = Mission.Current.Teams.GetAlliesOf(Mission.Current.Teams.Attacker, true);
+            var defender = Mission.Current.Teams.GetAlliesOf(Mission.Current.Teams.Defender, true);
+            
+            var agents =Mission.Current.Agents;
+            
+            
+            //assign Team allies to Parties
+
+            foreach (var team in attacker)
+            {
+                foreach (Agent agent in team.TeamAgents)
+                {
+                    if(agent.Character.IsPlayerCharacter||agent.Character.IsHero)
+                        TOWCommon.Say(agent.Name);
+                    
+                    if (agent.IsHero|| agent.IsMainAgent)
+                    {
+                        TOWCommon.Say(agent.Name);
+                    }
+                    else
+                    {
+                        TOWCommon.Say("agent: common enemy " );
+                    }
+                }
+
+               
+
+                
+            }
+            foreach (var team in defender)
+            {
+                foreach (Agent agent in team.TeamAgents)
+                {
+                    if(agent.Character.IsPlayerCharacter||agent.Character.IsHero)
+                        TOWCommon.Say(agent.Name);
+                    
+                    if (agent.IsHero||agent.IsMainAgent)
+                    {
+                        TOWCommon.Say("agent "+agent.Name);
+                    }
+                    else
+                    {
+                        TOWCommon.Say("agent: common enemy " );
+                    }
+                }
+
+               
+
+                
+            }
+        }
+
+        public override void AfterAddTeam(Team team)
+        {
+            base.AfterAddTeam(team);
+            
+        }
+
+        /*public override void OnAgentBuild(Agent agent, Banner banner)
+        {
+            base.OnAgentBuild(agent, banner);
+            
+            /*if (agent.IsHero)
+            {
+                TOWCommon.Say(defenderAttributes[0].WindsOfMagic.ToString());
+            }#1#
+        }*/
+
+        public override void OnRenderingStarted()
+        {
+            
+        }
+
+        public override void OnCreated()
+        {
+            
+            
+            
+        }
+
```

Comment:
This file has a lot of excessive new lines, please delete. Also, did you delete the commented block on 100-108? The conversation was resolved but the block still shows for me. Unpushed commit?

Reply:
Also, what are the empty OnCreated() and OnRenderingStarted() methods for?